### DISCUSSION TITLE
Investigate having R messages and possibly output in multiple languages

### DISCUSSION THREAD

**Discussion by rdstern** [IDEMSInternational/R-Instat#9767] [Climatic]
This is discussed in [the following presentation](https://www.youtube.com/watch?v=S3Qhqf9u6ao).

I wonder if it might now be easy to add to R-Instat?  The initial languages seem to include all we have used, except - of course - Swahili!

@lloyddewit could there be a feasibility study on whether this could be a "low-hanging fruit"?

**Comment by lloyddewit** [comment]
@rdstern 
Thank you for the link. The video is 1.5 hours so I just skipped through some key points.
I understand that R is able to display system messages in 16 different languages. 
For R-Instat, I assume that you're interested in displaying R system messages (errors, warnings etc.) in other languages (e.g. French). Did I understand correctly?

In R-Instat, I think R warnings and errors are always shown in message boxes. The message box may contain the R warning/error and some additional R-Instat text. R-Instat does not currently translate message boxes. So even if the R error message was in French, then currently the rest of the message box would still be in English.

According to [this link](https://stackoverflow.com/questions/13575180/how-to-change-language-settings-in-r), the R language can be changed by setting an environment variable. I tried this in the R-Instat script window but it had no effect. There may be something simple I am doing wrong or it may be a limitation of how R-Instat interacts with R. It would need investigating by someone with knowledge of R environments.

So I guess the next step is to determine the relative priority of this and decide if its worth asking an R developer to get this working on the command line. If that is possible then it should not be difficult for R-Instat to execute the necessary R command(s) whenever the language is changed.

![image](https://user-images.githubusercontent.com/57253949/211294841-5f69ae96-2e63-4d31-8bd4-2e6ceae895d1.png)
